### PR TITLE
New CFE workflow

### PR DESCRIPTION
**Pull request by naman03malhotra**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This workflow gets triggered when the CFE PR is merged into the release branch, cherry-picking the commit and creating a new PR against trunk.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

I am sure there must be a better way to test this, but these are the steps I followed:

Please disable some common workflows in your forked repo, like CI, etc, to save time.

1. Fork this repo in a different org, make sure the name is `woocommerce`.
2. Add this as a remote in your local.
3. Create a new branch `release/9.4` out of this branch and push that upstream in your new repo.
4. Create a regular fix - PR with changelog out of `release/9.4` as you usually do for `trunk`.
5. Squash and merge the PR in release branch.
6. It should trigger the new cherry-pick workflow, that will create a PR to merge the change in the `trunk` like [this](https://github.com/Alconomy/woocommerce/pull/6).
7. Note that workflow will fail due to slack workflow, but it shall work for `woocommerce` repo.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: .github/workflows/new-cfe-cherry-pick.yml [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        if: needs.verify.outputs.run == 'true'
+        needs: verify
+        outputs:
+            release: ${{ steps.prep-inputs.outputs.release }}
+            pr: ${{ steps.prep-inputs.outputs.pr }}
+            version: ${{ steps.prep-inputs.outputs.version }}
+        steps:
+            - name: Prep inputs
+              id: prep-inputs
+              uses: actions/github-script@v6
```
Comment (barryhughes) [inline .github/workflows/new-cfe-cherry-pick.yml]: Is it worth bumping this to `@v7`? Might head off warnings like this one:

> The following actions use a deprecated Node.js version and will be forced to run on node20:  
>  
> actions/github-script@v6.  
>  
> For more info: https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/

**Review by barryhughes — COMMENTED** [review]
Nice! Tested from a fork, and it worked as expected (a cherry-pick PR targeting `trunk` was quickly created):

<div align=center>Automatically generated cherry-pick PR</div>

Some notes/questions:

- I left a question about updating one of our actions, to eliminate a warning (and prevent future breakages).
- I couldn't successfully test the `Notify Slack on success|failure` steps, I guess because I was working from within my fork. The relevant yaml looks good, though.
- Curious about our plan for changelogs (specifically the changelog that is subsequently part of the cherry-pick PR). Are we handling that via a different PR?

**Comment by adimoldovan** [comment]
@naman03malhotra I wonder if this is a good time to also address the Github token issue described in https://github.com/woocommerce/woocommerce/issues/50375.

**Review by Stojdza — COMMENTED** [review]
The flow looks good and works as expected

More details about the review and findings are here: pf59Ax-20m-p2/#comment-1739

File: .github/workflows/new-cfe-cherry-pick.yml (hunk 1 above)
Comment (naman03malhotra) [reply .github/workflows/new-cfe-cherry-pick.yml]: Good catch, added it here: https://github.com/woocommerce/woocommerce/pull/51207/commits/739c2b7b6874ff032a71c734a8e56880bce4af72

**Comment by naman03malhotra** [comment]
>I wonder if this is a good time to also address the Github token issue described in https://github.com/woocommerce/woocommerce/issues/50375.

@adimoldovan good point, I think we can, if not as part of this PR, but we should do it as part of the new workflow that is targeted to be launched with 9.5 release.

Let me know that is needed to be done, and where can I help.

**Review by Stojdza — APPROVED** [review]
🚀

**Pull request merged**